### ISSUE TITLE
Paypal Sub Directory Multisite returns network_home_url  (2829)

### ISSUE THREAD

**Issue by djcowan** [woocommerce/woocommerce-paypal-payments#2114]
Clarification from WP support thread comment.
[wordpress support thread](https://wordpress.org/support/topic/return-url-not-correct/)

The return uri is correct when using express paypal checkout - the issue for us only appears when checkout is completed using woocommerce proceed to paypal button 
- Unvaulted , no Saved token for ppcp-gateway
- cancel and return url is correct

we changed `network_home_url()` to `home_url()` which successfully redirected - but we've only tasted a smol slice of pie.
let us know if additional info is requrest

![paypal-checkout](https://github.com/woocommerce/woocommerce-paypal-payments/assets/2137698/76d92d06-8f9b-4664-872e-b2be73a0c9b6)

--
https://github.com/woocommerce/woocommerce-paypal-payments/blob/09937fc991762c85af7afaf57d681ab253c17ea9/modules/ppcp-api-client/src/Repository/ApplicationContextRepository.php#L57

**Comment by InpsydeNiklas** [comment]
Hello @djcowan, thanks for providing additional information on this issue report. So far, we haven't been able to reproduce the behavior so far as the scenarios we tested on subsites in a Multisite network resulted in the proper `return_url` and `cancel_url` being logged:

`"return_url":"https:\/\/example.com\/site-2\/?wc-ajax=ppc-return-url","cancel_url":"https:\/\/example.com\/site-2\/checkout\/", `

The behavior appears to be consistent for the PayPal express & checkout buttons and also the "Proceed to PayPal" button, but we'll have a deeper look into this to better understand why in some cases the wrong URL is returned. 
Just for clarification, you are seeing the URL from the main site of the network as the `return_url` on the subsite, did I get that right? Thank you!

**Comment by djcowan** [comment]
Yes, correct. We are seeing the main site url as the PayPal return_url and not the sub directory site domain

**Comment by floek** [comment]
Same issue here.
My multisite network is setup with domains instead of sub-directories.

**Comment by floek** [comment]
Any updates here? I'm still in my hand-patched fix.

**Comment by hookloom** [comment]
Hi, we also had a similar issue and reported it via a ticket when working on something else. Same line of code was the problem. For us, we have a subdirectory installation for our dev site at domain.com/dev, and the return_url is domain.com/dev/dev/...

We're working around it by hooking into the home_url and removing the duplicate folder if detected. (One can't hook into network_home_url because it can return home_url without even going through the filter).

**Comment by InpsydeNiklas** [comment]
This was addressed in #2157 and is included in the upcoming [2.7.0 update](https://github.com/woocommerce/woocommerce-paypal-payments/releases). Thanks for the feedback!

**Comment by darkopetreski** [comment]
I am experiencing the same error with 2.9.0 version.
The return url is domain.com/shop/shop/?.... and should be domain.com/shop?....
The problem occurs when I disable the smart button and do it with redirect:

```
add_filter('woocommerce_paypal_payments_use_place_order_button', '__return_true');
add_filter('woocommerce_paypal_payments_place_order_button_text', function() {
    return __( 'Place order', 'woocommerce' );
});
```

Edit: I have created a new issue: https://github.com/woocommerce/woocommerce-paypal-payments/issues/2604
